tools/SourceKit/include/SourceKit/Core/LangSupport.h

```diff
@@ -203,6 +203,7 @@ struct DiagnosticEntryInfoBase {
   std::string Filename;
   SmallVector<std::pair<unsigned, unsigned>, 2> Ranges;
   SmallVector<Fixit, 2> Fixits;
+  SmallVector<std::string, 1> EducationalNotes;
```

Comment:
Can these attach to notes or just to primary diagnostics?  If it's just primary diagnostics we could move this down to `DiagnosticEntryInfo`.

Reply:
These can be attached to notes, although nothing utilizes it yet. I'd like to continue supporting it for now, I think it could be useful for some of the diagnostics that offer multiple fix-it choices via notes.

Reply:
Okay cool. The implementation here is fine then.  When you get to this point, you might want to look at how notes and fixits are being surfaced in LSP as "DiagnosticRelatedInformation" and "CodeActions".  The compiler's "note" cover both of these cases, but they really behave differently in the editor. In LSP a note like "look here" is related information and a code action is a note representing a fixit.  There is no overlap at the API level.  I wonder if we should preserve that distinction in the compiler as well.